### PR TITLE
Medical record

### PR DESCRIPTION
**Pull request by LRUki**
Add a new pallet **medical-record** to support functionalities for users (patient or doctor) ro
* Create an account
* Post record (Verified record for doctor, unverified record for patient)

Note that the records are stored in a double map
```
(AccountId, UserType) => Record[]
```

Unit tests done in tests.rs

### CODE REVIEW

File: pallets/medical-record/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+[dependencies]
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = [
+	"derive",
+] }
+scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
+frame-benchmarking = { version = "4.0.0-dev", default-features = false, optional = true, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.32" }
+frame-support = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.32" }
+frame-system = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.32" }
+sp-runtime = { version = "6.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.32" }
+sp-std = { git = "https://github.com/purestake/substrate", branch = "moonbeam-polkadot-v0.9.29", default-features = false }
```
Comment (OverOrion) [inline pallets/medical-record/Cargo.toml]: I believe it should be this:
```suggestion
sp-std = { version = "4.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.32" }
```

File: pallets/medical-record/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+[package]
+name = "pallet-medical-record"
+version = "4.0.0-dev"
+description = "FRAME pallet template for defining custom runtime logic."
+authors = ["Substrate DevHub <https://github.com/substrate-developer-hub>"]
+homepage = "https://substrate.io"
+edition = "2021"
+license = "Unlicense"
+publish = false
+repository = "https://github.com/substrate-developer-hub/substrate-node-template/"
```
Comment (OverOrion) [inline pallets/medical-record/Cargo.toml:10]: [Note]
This should be edited accordingly (not necessarily now, but later on).

File: pallets/medical-record/src/tests.rs [hunk 3]
```diff
@@ -59,3 +46,11 @@ fn patient_can_add_record() {
 		);
 	});
 }
+
+fn origin_to_account_id(o: RuntimeOrigin) -> u64 {
+	let ro: Result<RawOrigin<u64>, RuntimeOrigin> = o.clone().into();
+	match ro.ok().unwrap() {
```
Comment (OverOrion) [inline pallets/medical-record/src/tests.rs]: [Nitpicking]
Could you please change the `unwrap()` to a `expect()` just for the sake of clarity?

File: pallets/medical-record/src/tests.rs [hunk 4]
```diff
@@ -45,20 +49,53 @@ fn patient_can_add_record() {
...
+				doctor.clone(),
+				origin_to_account_id(patient.clone()),
+				BoundedVec::with_max_capacity(),
+				BoundedVec::with_max_capacity(),
+			));
+		}
+
+		let patient_records =
+			MedicalRecord::records(origin_to_account_id(patient.clone()), UserType::Patient)
+				.unwrap();
```
Comment (OverOrion) [inline pallets/medical-record/src/tests.rs]: [Nitpicking]
Could you please change the unwrap() to a expect() just for the sake of clarity?

File: pallets/medical-record/src/tests.rs [hunk 5]
```diff
@@ -35,7 +39,7 @@ fn patient_can_add_record() {
 
 		let records =
 			MedicalRecord::records(origin_to_account_id(o.clone()), UserType::Patient).unwrap();
```
Comment (OverOrion) [inline pallets/medical-record/src/tests.rs]: [Nitpicking]
Could you please change the unwrap() to a expect() just for the sake of clarity?

**Review by OverOrion — COMMENTED** [review]
Overall LGTM, I have only one thing which needs to be changed before merging it.
Let me know if you want me to apply my suggestions and merge it.

File: pallets/medical-record/Cargo.toml (hunk 1 above)
Comment (LRUki) [reply pallets/medical-record/Cargo.toml]: Good catch. Was looking into the example Andrew gave and forgot to change!

> 
Yea feel free to fix 


> Overall LGTM, I have only one thing which needs to be changed before merging it.
> Let me know if you want me to apply my suggestions and merge it.

Yea feel free to apply the change and merge it!

**Review by OverOrion — APPROVED** [review]
LGTM, thank you!

**Pull request merged**